src/cascadia/ShellExtension/pch.h

```diff
@@ -23,7 +23,7 @@
 
 #include <winrt/base.h>
 
-#include <Unknwn.h>
+#include <unknwn.h>

```

Comment:
Most of the repository uses the lowercase name. I'm not quite sure what the story is for this file, but on most Windows computers the file system is case-preserving and case-insensitive so it doesn't matter. It is possible to mount NTFS as case-sensitive in which case a mismatch would cause a failure.

Reply:
I should note that while this repository mostly spells this file name in lowercase, most instances of this file reference in `microsoft/PowerToys` has it with the initial `U`... I don't have an SDK handy, so I'm not sure which is canonical, but if someone could check, I'd appreciate it.

Reply:
![image](https://github.com/user-attachments/assets/425082d1-9998-4a6c-b1a3-38cad9410dc8)

It looks like capital-`U` is technically canonical. It's published internally as `unknwn` but staged into the SDK as `Unknwn`.

Clang has a warning for this, but it does not emit by default (even with its most picky warning modes!) unless you explicitly request it.

![image](https://github.com/user-attachments/assets/62f98ba1-b4ff-4ec9-9a3a-f0940c467705)

All that to say...

I'm OK leaving it `unknwn` with the lower-`u`.

Reply:
Hah. I had a vague recollection that it was `Unknwn`. I'd be fine w/ fixing all of the cases in this repository to `Unknwn` instead of `unknwn` since that's the official spelling (even if the internal case is more normal).

Reply:
https://github.com/microsoft/terminal/compare/c0914074dbef9475db0408dc6e288f2ec391436b..8f7f728213868a97fb3b5facac0e4988f03cef42

Reply:
I really should get a copy of the Windows SDK on GitHub. I want to know if `Windows.h` or `windows.h` is correct, and I also want to know why this file has `unknwn.h` twice in close succession:

https://github.com/microsoft/PowerToys/blob/54aab5d10954c53d78143eddb1b65c114f93e472/src/modules/powerrename/PowerRenameUILib/pch.h#L5-L9
(PowerToys is split 2:1 on case for windows.h:Windows.h)
(Terminal is split 4:1 on case for windows.h:Windows.h)